### DISCUSSION TITLE
SourceMapDevToolPlugin not creating source maps for all files

### DISCUSSION THREAD

**Discussion by ssundahlTTD** [webpack/webpack#13810] [Q&A]
I didn't want to log this as an issue in case I'm doing something wrong.  But I'm trying to mimic devtool: 'source-map' behavior using SourceMapDevToolPlugin.  I read somewhere that devtool: 'source-map' is the same as 

```
    devtool: false,     
    plugins: [
        new webpack.SourceMapDevToolPlugin({
            filename: '[file].map',
        }),
    ],
```

When I run devtool: 'source-map' I get an actual source map for every file.  When I use the other method I get less than half of the source maps with no obvious (to me) pattern.  The reason I'm trying to do this is that source map generation for 1000+ files takes forever and I want to use include/exclude to only generate maps for files that I'll likely be interested in debugging in production.

Thanks in advance.

![image](https://user-images.githubusercontent.com/37344964/125978317-087f4f50-2184-4466-b1bc-c1ff88d65b18.png)

**Comment by alexander-akait** [comment]
Can you provide example?

**Reply by ssundahlTTD** [reply]
aside from the screenshot above?  That's an example of the random set of files that actually get a source map.  When I just do devtool: 'source-map' everything gets a source map.

**Reply by alexander-akait** [reply]
but it is expected, maybe I am missing something...

**Reply by ssundahlTTD** [reply]
It's expected that only some files get source maps?

**Reply by alexander-akait** [reply]
Do you need source map for CSS?

**Reply by ssundahlTTD** [reply]
Not at all.  But source maps are generated for CSS files when I use devtool: 'source-map'.  I'm trying to understand why the two different approaches are yielding different results.  As you can see from the screenshot above, a number of javascript files are also not getting source maps.  I'd like to understand why that is.  Without knowing how to reliably generate maps for the files I want, the plugin is useless to me.

**Reply by alexander-akait** [reply]
Because `devtool: 'source-map'` say to all loaders/plugins to generate source map, but `webpack.SourceMapDevToolPlugin` is low level api

**Reply by ssundahlTTD** [reply]
Is there a combination of settings to make webpack.SourceMapDevToolPlugin do what devtool: 'source-map' does?  I'd really love to understand what it's doing to decide which files to generate.  Because it looks random to me.  Out of something like 1500 files it only generated 167 source maps.

**Reply by alexander-akait** [reply]
Yep, here our logic https://github.com/webpack/webpack/blob/main/lib/WebpackOptionsApply.js#L190
